safe/impact_function/impact_function.py

```diff
@@ -1997,6 +1996,8 @@ def provenance(self):
         If the impact function is not ready (has not called prepare method),
         it will return empty dict to avoid miss information.
 
+        The impact function will call generate_provenance at the end of the IF.
+
         List of keys (for quick lookup):
 
         [
```

Comment:
@Gustry and @ismailsunni It's hard to keep track what can be expected to be inside the provenance for now.
So, if possible, if there's any changes in the future, please change this list accordingly (for example, we might want to delete report_question?).
Best solution would be to declare another container class for provenance (for self._provenance object).
That way, pycharm can detect its properties without having to run it.